manifests/service.pp

```diff
@@ -21,6 +21,17 @@
   $install_dir       = $wso2base::install_dir
   $carbon_home       = $wso2base::carbon_home
   $autostart_service = $wso2base::autostart_service
+  $file_list_copy_without_refresh = $wso2base::file_list_copy_without_refresh
+
+  if ($file_list_copy_without_refresh != undef and size($file_list_copy_without_refresh) > 0) {
+    wso2base::push_files {
+      $file_list_copy_without_refresh:
+        owner       => $wso2_user,
+        group       => $wso2_group,
+        carbon_home => $carbon_home,
+        wso2_module => $caller_module_name
```

Comment:
According to [1] by default puppet applies resources in the order they are declared in the manifest. The execution order would be wso2base::push_files -> Service[$service_name] -> notify. According to my understanding, this is not required.

[1] https://docs.puppet.com/puppet/latest/lang_relationships.html